app/src/main/kotlin/com/github/gotify/messages/IntentUrlDialogActivity.kt

```diff
@@ -0,0 +1,40 @@
+package com.github.gotify.messages
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.github.gotify.R
+import com.github.gotify.databinding.ActivityDialogIntentUrlBinding
+
+internal class IntentUrlDialogActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setFinishOnTouchOutside(false)
+        val binding = ActivityDialogIntentUrlBinding.inflate(layoutInflater)
+        val intentUrl = intent.getStringExtra(EXTRA_KEY_URL)
+        binding.cancelButton.setOnClickListener { finish() }
+
+        if (intentUrl == null) {
+            binding.urlView.text = getString(R.string.action_dialog_missing)
+            binding.openButton.visibility = View.GONE
+            setContentView(binding.root)
+            return
+        }
```

Comment:
I think this cannot happen, so we don't need to handle this with good UX. An assert should be enough imo. What do you think?
```
        assert(intentUrl != null) { "intentUrl may not be empty" }
```

Reply:
Just wanted to be safe here, but we can also adopt your version.  
Changed in d6114359c60c66400d5e74b465ae7a51dd58f83d